### PR TITLE
fix(query) Pushdown in SinglePartitionPlanner

### PR DESCRIPTION
**Pull request by amolnayak311**
**Pull Request checklist**

- [x] The commit(s) message(s) follows the contribution [guidelines](CONTRIBUTING.md) ?
- [x] Tests for the changes have been added (for bug fixes / features) ?
- [ ] Docs have been added / updated (for bug fixes / features) ?

**Current behavior :** 

Currently the high level ``LogicalPlan``  is used to decide on whether to perform the Inprocess operation of delegate to one of the underlying planner instances. This causes operations like 

``sum(foo{} + bar:1m{})`` to be wrongly pushed to raw planner. Instead the data needs to be joined in process by pulling data from both raw and RR cluster

**New behavior :**

The plan is walked and only at appropriate level, the underlying planner is used to materialize the ``LogicalPlan``

**BREAKING CHANGES**

Places where ``SinglePartitionPlanner`` is instantiated needs to change. We will need to pass ``DataSet`` instead of the metrics name.

### CODE REVIEW

File: coordinator/src/main/scala/filodb.coordinator/queryplanner/SinglePartitionPlanner.scala [hunk 1]
```diff
@@ -1,131 +1,133 @@
...
+   * inprocess are defined in DefaultPlanner and thus we delegate the materialization of these plan to the default
+   * walk implementation in DefaultPlanner
+   *
+   * @param logicalPlan    The LogicalPlan instance
+   * @param qContext       The QueryContext
+   * @param forceInProcess if true, all materialized plans for this entire
+   *                       logical plan will dispatch via an InProcessDispatcher
+   * @return The PlanResult containing the ExecPlan
+   */
+  override def walkLogicalPlanTree(logicalPlan: LogicalPlan, qContext: QueryContext, forceInProcess: Boolean)
```
Comment (alextheimer) [inline coordinator/src/main/scala/filodb.coordinator/queryplanner/SinglePartitionPlanner.scala:95]: A+ for this method alone-- thanks for cleaning everything up and getting rid of that copy-paste code from the `DefaultPlanner`.

File: coordinator/src/test/scala/filodb.coordinator/queryplanner/PlannerHierarchySpec.scala [hunk 2]
```diff
@@ -855,4 +853,294 @@ class PlannerHierarchySpec extends AnyFunSpec with Matchers with PlanValidationS
...
+                         |----E~MultiSchemaPartitionsExec(dataset=timeseries, shard=0, chunkMethod=TimeRangeChunkScan(1633913030000,1634777330000), filters=List(ColumnFilter(_ws_,Equals(demo)), ColumnFilter(_ns_,Equals(localNs)), ColumnFilter(instance,Equals(Inst-1)), ColumnFilter(_metric_,Equals(foo:1m))), colName=None, schema=None) on ActorPlanDispatcher(Actor[akka://default/system/testProbe-1#-913106743],recordingRules)
+                         |---T~PeriodicSamplesMapper(start=1633913330000, step=300000, end=1634777330000, window=None, functionId=None, rawSource=true, offsetMs=None)
+                         |----E~MultiSchemaPartitionsExec(dataset=timeseries, shard=1, chunkMethod=TimeRangeChunkScan(1633913030000,1634777330000), filters=List(ColumnFilter(_ws_,Equals(demo)), ColumnFilter(_ns_,Equals(localNs)), ColumnFilter(instance,Equals(Inst-1)), ColumnFilter(_metric_,Equals(foo:1m))), colName=None, schema=None) on ActorPlanDispatcher(Actor[akka://default/system/testProbe-1#-913106743],recordingRules)""".stripMargin
+    validatePlan(execPlan, expectedPlan)
+  }
+
+
+  it("Scalar fixed double plan with a Recording rule should be materialized in RR cluster") {
```
Comment (alextheimer) [inline coordinator/src/test/scala/filodb.coordinator/queryplanner/PlannerHierarchySpec.scala:1060]: Should we have tests for each of these?
```
foo > scalar(bar:1m)
foo:1m > scalar(bar)
```

**Review by alextheimer — APPROVED** [review]
Talk about a quick turnaround time. Thanks for the PR (and especially all the tests)-- minor comments added.

File: coordinator/src/test/scala/filodb.coordinator/queryplanner/PlannerHierarchySpec.scala (hunk 2 above)
Comment (amolnayak311) [reply coordinator/src/test/scala/filodb.coordinator/queryplanner/PlannerHierarchySpec.scala:1060]: Done, added these tests.

**Pull request merged**